src/pybind/mgr/dashboard_v2/controllers/pool.py

```diff
@@ -47,3 +49,40 @@ def list(self, attrs=None, stats=False):
     def get(self, pool_name, attrs=None, stats=False):
         pools = self.list(attrs, stats)
         return [pool for pool in pools if pool['pool_name'] == pool_name][0]
+
+    # pylint: disable=too-many-arguments, too-many-locals
+    @RESTController.args_from_json
+    def create(self, pool, pg_num, pool_type, erasure_code_profile=None, cache_mode=None,
+               tier_of=None, read_tier=None, flags=None, compression_required_ratio=None,
+               application_metadata=None, crush_rule=None, rule_name=None, **kwargs):
+        ecp = erasure_code_profile if erasure_code_profile else None
+        CephService.send_command('mon', 'osd pool create', pool=pool, pg_num=int(pg_num),
+                                 pgp_num=int(pg_num), pool_type=pool_type, erasure_code_profile=ecp)
+
+        if cache_mode:
+            CephService.send_command('mon', 'osd tier cache-mode', pool=pool, mode=cache_mode)
+        if tier_of:
+            CephService.send_command('mon', 'osd tier add', tierpool=tier_of, pool=pool)
+        if read_tier:
+            CephService.send_command('mon', 'osd tier set-overlay', pool=pool,
+                                     overlaypool=read_tier)
+        if flags and 'ec_overwrites' in flags:
+            CephService.send_command('mon', 'osd pool set', pool=pool, var='allow_ec_overwrites',
+                                     val='true')
+        if compression_required_ratio:
+            CephService.send_command('mon', 'osd pool set', pool=pool,
+                                     var='compression_required_ratio',
+                                     val=str(compression_required_ratio))
+        if application_metadata:
+            for app in set(json.loads(application_metadata)):
+                CephService.send_command('mon', 'osd pool application enable', pool=pool, app=app)
+        if crush_rule:
+            CephService.send_command('mon', 'osd pool set', pool=pool, var='crush_rule',
+                                     val=rule_name)
+        for key, value in kwargs.items():
+            if type == 'replicated' and key not in \
+                    ['name', 'erasure_code_profile_id'] and value is not None:
+                CephService.send_command('mon', 'osd pool set', pool=pool, var=key, val=value)
```

Comment:
Isn't this code a bit of dangerous? basically you are calling `osd pool set` for unknown (key, value) pairs that come from `kwargs` without any validation.
Since, the frontend is the consumer of this API we are in controll of all things that come inside the request JSON, therefore we should validate them in the backend.

Reply:
We have validation in C++ and I'm trusting the C++ validation. And if something is not properly validated, I'd favor fixing the C++ validation.

Reply:
OK, I guess this is part of a more generic problem of handling the errors from Ceph library calls, which we still haven't tackled properly.

Reply:
Indeed. let's tackle this after this PR.